wrapper/src/main/java/software/amazon/jdbc/util/monitoring/AbstractMonitor.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package software.amazon.jdbc.util.monitoring;
+
+import java.util.logging.Logger;
+import software.amazon.jdbc.util.Messages;
+
+public abstract class AbstractMonitor implements Monitor, Runnable {
```

Comment:
Does it make sense to add heartbeat functionality to this class? It'd be nice just to inherit it.

Reply:
That would be nice, I'm not sure how to add that though. Each monitor type will have its own custom run loop that it needs to define, and the last updated timestamp needs to be updated somewhere in that custom loop. Not sure how to extract the heartbeat piece to the abstract monitor while leaving the custom loop implementation to the underlying monitor class.